searchlib/src/vespa/searchlib/test/attribute_builder.cpp

```diff
@@ -0,0 +1,131 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "attribute_builder.h"
+#include <vespa/searchlib/attribute/attributefactory.h>
+#include <vespa/searchlib/attribute/attributevector.h>
+#include <vespa/searchlib/attribute/integerbase.h>
+#include <vespa/searchlib/attribute/stringbase.h>
+#include <cassert>
+
+namespace search::attribute::test {
+
+AttributeBuilder::AttributeBuilder(const vespalib::string& name, const Config& cfg)
+    : _attr_ptr(AttributeFactory::createAttribute(name, cfg)),
+      _attr(*_attr_ptr)
+{
+}
+
+AttributeBuilder::AttributeBuilder(AttributeVector& attr)
+    : _attr_ptr(),
+      _attr(attr)
+{
+}
+
+namespace {
+
+void
+add_docs(AttributeVector& attr, size_t num_docs)
+{
+    attr.addReservedDoc();
+    attr.addDocs(num_docs);
+}
+
+template <typename AttrType, typename ValueType>
+void
+fill_helper(AttributeVector& attr, const std::vector<ValueType>& values)
+{
+    assert(attr.getConfig().collectionType() == CollectionType::SINGLE);
+    add_docs(attr, values.size());
+    auto* real = dynamic_cast<AttrType*>(&attr);
```

Comment:
Consider using `dynamic_cast<AttrType&>(attr)`, which throws an exception if cast cannot be performed.